### PR TITLE
feat: pkg(kcl schema to go and go struct to kcl schema)

### PR DESCRIPTION
**Pull request by huangch96**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [X] Y 

re https://github.com/KusionStack/KCLVM/issues/198

#### 2. What is the scope of this PR (e.g. component or file name):

kclvm-go/pkg/gogen
kclvm-go/pkg/kclgen

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [X] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [X] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [X] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by paragon-h** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Peefy** [comment]
Thank you for your contribution. Perhaps adding more annotations and examples to public functions and constants would be a better choice.

File: pkg/tools/gogen/gogen.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	genStruct := make(GenStruct, 0)
+	for _, kt := range ktList {
+		if kt.Type == "schema" {
+			schemaName = StructName(kt.SchemaName)
+			var schemaList []*Field
+			for k, p := range kt.Properties {
+				var field Field
+				field.Name = strings.ToUpper(k[:1]) + k[1:]
+				switch p.Type {
+				case "schema":
```
Comment (Peefy) [inline pkg/tools/gogen/gogen.go:35]: Perhaps the `any` type was missed

File: pkg/tools/gogen/gogen.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+					field.SchemaType = "int"
+					field.Tag = fmt.Sprintf(`kcl:"name=%s,type=%s"`, k, "int")
+				case "float":
+					field.SchemaType = "float32"
+					field.Tag = fmt.Sprintf(`kcl:"name=%s,type=%s"`, k, "float")
+				case "bool":
+					field.SchemaType = "bool"
+					field.Tag = fmt.Sprintf(`kcl:"name=%s,type=%s"`, k, "bool")
+				case "null":
+					field.SchemaType = "nil"
```
Comment (Peefy) [inline pkg/tools/gogen/gogen.go:71]: nil type or any type or `interface{}`?

File: pkg/tools/gogen/gogen.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	for _, kt := range ktList {
+		if kt.Type == "schema" {
+			schemaName = StructName(kt.SchemaName)
+			var schemaList []*Field
+			for k, p := range kt.Properties {
+				var field Field
+				field.Name = strings.ToUpper(k[:1]) + k[1:]
+				switch p.Type {
+				case "schema":
+					field.SchemaType = "*" + p.SchemaName
```
Comment (Peefy) [inline pkg/tools/gogen/gogen.go:36]: For the `field.SchemaType`, how about using the code defined here: https://github.com/KusionStack/kclvm-go/blob/main/pkg/tools/gen/gengo.go#L109

Comment (alanhuangch) [reply pkg/tools/gogen/gogen.go:36]: ok，got it. I find that you have commited the gengo code. My branch was behind from the commit code. I will use the code `GetTypeName`.

File: pkg/tools/gogen/gogen.go (hunk 2 above)
Comment (alanhuangch) [reply pkg/tools/gogen/gogen.go:71]: I don't know this usage about kcl schema 
```
schema Movie:
	desc: str = ""
	size: NumberMultiplier = 2M
	kind?: "Superhero" | "War" | "Unknown"
	unknown1?: int | str = None
	unknown2?: any = None
```
will fix it

**Comment by Peefy** [comment]
Thanks @ekkoful. Pls rebase the main branch and remove the `gogen` package, then I will merge this PR with the `kclgen` package.

**Comment by paragon-h** [comment]
> Thanks @ekkoful. Pls rebase the main branch and remove the `gogen` package, then I will merge this PR with the `kclgen` package.

ok, I have try a new way to generate the KCL schema code today. When I implement it, I will commit a new PR for this issue. Then I will close the PR. Thanks @Peefy . You give me a lot  inspire from your code.

**Comment by paragon-h** [comment]
> > Thanks @ekkoful. Pls rebase the main branch and remove the `gogen` package, then I will merge this PR with the `kclgen` package.
> 
> ok, I have try a new way to generate the KCL schema code today. When I implement it, I will commit a new PR for this issue. Then I will close the PR. Thanks @Peefy . You give me a lot inspire from your code.

The current code, Only can use struct interface for func args. And I implemented via the package https://github.com/fatih/structs. Now I want to use `go/parser` `go/ast` to parse go source code file. Maybe it will be a better way.

**Comment by Peefy** [comment]
> > > Thanks @ekkoful. Pls rebase the main branch and remove the `gogen` package, then I will merge this PR with the `kclgen` package.
> > 
> > 
> > ok, I have try a new way to generate the KCL schema code today. When I implement it, I will commit a new PR for this issue. Then I will close the PR. Thanks @Peefy . You give me a lot inspire from your code.
> 
> The current code, Only can use struct interface for func args. And I implemented via the package https://github.com/fatih/structs. Now I want to use `go/parser` `go/ast` to parse go source code file. Maybe it will be a better way.

OK, come on.

**Comment by Peefy** [comment]
Closed this PR and opened PR #97

**Pull request closed without merging**